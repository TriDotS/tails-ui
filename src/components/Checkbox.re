let component = ReasonReact.statelessComponent("Checkbox");

let make = (~checked=false, ~disabled=false, ~label=?, _children) => {
  ...component,
  render: _self => {
    let labelComponent =
      switch label {
      | None => ReasonReact.nullElement
      | Some(label) => ReasonReact.stringToElement(label)
      };
    <div className="flex items-center">
      <input
        _type="checkbox"
        checked=(Js.Boolean.to_js_boolean(checked))
        disabled=(Js.Boolean.to_js_boolean(disabled))
      />
      (
        labelComponent === ReasonReact.stringToElement("") ?
          <label
            className="block uppercase tracking-wide text-grey-darker text-xs font-bold mb-2">
            labelComponent
          </label> :
          ReasonReact.nullElement
      )
    </div>;
  }
};

let default =
  ReasonReact.wrapReasonForJs(~component, jsProps =>
    make(
      ~checked=jsProps##checked,
      ~disabled=jsProps##disabled,
      ~label=jsProps##label,
      [||]
    )
  );
