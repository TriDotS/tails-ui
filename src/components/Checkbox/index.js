import PropTypes from 'prop-types'
import React from 'react'

const Checkbox = ({ checked, disabled, handleChange, label }) => (
  <div className="flex items-center">
    <input
      type="checkbox"
      checked={checked}
      disabled={disabled}
      onChange={handleChange}
    />
    {label && <label className="ml-2 text-sm">{label}</label>}
  </div>
)

Checkbox.propTypes = {
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  handleChange: PropTypes.func,
  label: PropTypes.string
}

export default Checkbox
