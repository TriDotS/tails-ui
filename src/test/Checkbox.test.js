import Checkbox from '../components/Checkbox/Checkbox'
import React from 'react'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(<Checkbox />, div)
})

test('checkbox snapshot', () => {
  const tree = renderer.create(<Checkbox />).toJSON()
  expect(tree).toMatchSnapshot()
})
