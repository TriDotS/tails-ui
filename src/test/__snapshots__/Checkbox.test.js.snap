// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox snapshot 1`] = `
<div
  className="flex items-center"
>
  <input
    checked={undefined}
    disabled={undefined}
    onChange={undefined}
    type="checkbox"
  />
</div>
`;
